attestor/policy/attestation_policy.go

```diff
@@ -104,9 +174,133 @@ func CheckPreventBinaryArtifacts(
 		}
 	}
 
+	dl.Info(&checker.LogMessage{Text: "repo was free of binary artifacts"})
 	return Pass, nil
 }
 
+func CheckNoVulnerabilities(results *checker.RawResults, dl checker.DetailLogger) (PolicyResult, error) {
+	nVulns := len(results.VulnerabilitiesResults.Vulnerabilities)
+
+	dl.Info(&checker.LogMessage{Text: fmt.Sprintf("found %d vulnerabilities in package", nVulns)})
+
+	return nVulns == 0, nil
+}
+
+func toString(cs checker.Changeset) string {
+	platform := cs.ReviewPlatform
+	if platform == "" {
+		platform = "unknown"
+	}
+	return fmt.Sprintf("%s(%s)", platform, cs.RevisionID)
+}
+
+func CheckCodeReviewed(
+	reqs CodeReviewRequirements,
+	results *checker.RawResults,
+	dl checker.DetailLogger,
+) (PolicyResult, error) {
+	for _, changeset := range results.CodeReviewResults.DefaultBranchChangesets {
+		numApprovers := 0
+		approvers := make(map[string]bool)
+		// CodeReview check is limited to github.com pull request reviews
+		// Log if a change isn't a github pr since it's a bit unintuitive
+		foundLinkedReviews := false
+
+		for _, appr := range reqs.RequiredApprovers {
+			// Required approvers must approve, other approvals optional
+			approvers[appr] = false
+		}
+
+		for i := range changeset.Commits {
+			commit := changeset.Commits[i]
+			for _, review := range commit.AssociatedMergeRequest.Reviews {
+				foundLinkedReviews = true
+				if review.State == "APPROVED" {
+					numApprovers += 1
+					approvers[review.Author.Login] = true
+				}
+			}
+		}
+
+		if !foundLinkedReviews {
+			dl.Warn(
+				&checker.LogMessage{
+					Text: fmt.Sprintf(
+						"no code reviews linked to %s",
+						toString(changeset),
+					),
+				},
+			)
+		}
+
+		if numApprovers < reqs.MinReviewers {
+			dl.Info(
+				&checker.LogMessage{
+					Text: fmt.Sprintf(
+						"not enough approvals for %s (needed:%d found:%d)",
+						toString(changeset),
+						reqs.MinReviewers,
+						numApprovers,
+					),
+				},
+			)
+			return Fail, nil
+		}
+
+		missingApprovers := false
+		for appr, present := range approvers {
```

Comment:
nit: follow-up of nit above. 
You could just iterate over `reqs.RequiredApprovers` instead of the entire `approvers` map.